src/code/client.coffee

```diff
@@ -103,6 +103,38 @@ class CloudFileManagerClient
   connect: ->
     @_event 'connected', {client: @}
 
+  #
+  # Called from CloudFileManager.clientConnect to process the URL parameters
+  # and initiate opening any document specified by URL parameters. The CFM
+  # hash params are processed here after which providers are given a chance
+  # to process any provider-specific URL parameters. Calls ready() if no
+  # initial document opening occurs.
+  #
+  processUrlParams: ->
+    # process the hash params
+    hashParams = @appOptions.hashParams
+    if hashParams.sharedContentId
+      @openSharedContent hashParams.sharedContentId
+    else if hashParams.fileParams
+      if hashParams.fileParams.indexOf("http") is 0
+        @openUrlFile hashParams.fileParams
+      else
+        [providerName, providerParams] = hashParams.fileParams.split ':'
+        @openProviderFile providerName, providerParams
+    else if hashParams.copyParams
+      @openCopiedFile hashParams.copyParams
+    else
+      # give providers a chance to process url params
+      for provider in @state.availableProviders
+        handleResult = provider.canHandleUrlParams()
+        if handleResult and handleResult.providerName
+          {providerName, providerParams} = handleResult
+          @openProviderFile providerName, providerParams
+          return
```

Comment:
`canHandleUrlParams` sounds like a generic ability that the provider can choose to act on any arbitrary urlParams, but here it seems to be used simply as "can we call `openProviderFile` with some value?"

Couldn't the provider itself decide what to do with those params? It could call back to `client. openProviderFile`, or it could do something else entirely. Then you could even put the share params in the DocumentServer's `canHandleUrlParams`. (Maybe it would be renamed to `handleUrlParams`.)